### DISCUSSION TITLE
Simulating REM data feeds

### DISCUSSION THREAD

**Discussion by CodeSlinger69** [tagyoureit/nodejs-poolController#817] [REM] [answered]
@tagyoureit @rstrouse I'm doing some testing and experimenting as you know, and thus far I've been able to test, trace, and simulate things pretty well with VIsual Studio Code's debugger, and Chrome's debugger. But I'm at the point now where I would like to simulate REM data feeds into njsPC. For example, set the air temperature to my test value, change it to another value, and so on.

How do you guys do your testing of REM data feeds and the effect on njsPC logic? Have you set up a REM simulator or similar?

**Comment by Crewski** [comment, marked as answer]
For temperatures, you can use Postman to do a PUT request.  The URL is `http://<ip_address>:4200/state/temps`.  The payload is a json with your values (i.e. `{"airSensor": 74, "waterSensor1": 85}`)

**Reply by rstrouse** [reply]
Its even easier than that.  If you run REM on Windows it will simulate all of the device types and even randomize the device outputs.  You can even try out mock devices for triggers, feeds, and attribute settings.

**Reply by CodeSlinger69** [reply]
Postman works great, thanks for that pointer.

@rstrouse Can you explain how to do this simulation? I just set up a REM instance running on a Windows 10 VM, and I don't see anything different in REM's web interface.

**Reply by rstrouse** [reply]
Go ahead and add an I2c device for EZO-RTD (after you add a bus).  This is the quickest way to have an temp feed.  You can also add and ADC or any of the other devices.

**Reply by CodeSlinger69** [reply]
Very cool, thank you. I did have to click "Reset Server" after adding the i2c bus and before adding devices, in case anyone else wants to try this.

This does indeed let you generate mock devices in REM, but only when running on Windows. I tried the same on Linux and no luck. I'm curious about that, but that's for another day.

The mock devices seem to generate random outputs, as you mentioned above, so if I want to run some tests with specific values, Postman looks like the way to go.

Still, good to have both of these tools in our toolbox, thanks guys.
